apps/server/src/modules/board/service/drawing-do-authorizable.service.spec.ts

```diff
@@ -0,0 +1,209 @@
+	let service: DrawingDoAuthorizableService;
+	let drawingDoRepo: DeepMocked<DrawingDoRepo>;
+	let courseRepo: DeepMocked<CourseRepo>;
+
+	beforeAll(async () => {
+		module = await Test.createTestingModule({
+			providers: [
+				DrawingDoAuthorizableService,
+				{
+					provide: DrawingDoRepo,
+					useValue: createMock<DrawingDoRepo>(),
+				},
+				{
+					provide: CourseRepo,
+					useValue: createMock<CourseRepo>(),
+				},
+			],
+		}).compile();
+
+		service = module.get(DrawingDoAuthorizableService);
+		drawingDoRepo = module.get(DrawingDoRepo);
+		courseRepo = module.get(CourseRepo);
+		await setupEntities();
+	});
+
+	afterAll(async () => {
+		await module.close();
+	});
+
+	describe('findById', () => {
+		describe('when finding a board domainobject', () => {
+			const setup = () => {
+				const columnBoard = columnBoardFactory.build();
+				drawingDoRepo.findById.mockResolvedValue(columnBoard);
+
+				return { columnBoardId: columnBoard.id };
+			};
+
+			it('should call the repository', async () => {
+				const { columnBoardId } = setup();
+
+				await service.findById(columnBoardId);
+
+				expect(drawingDoRepo.findById).toHaveBeenCalledWith(columnBoardId, 1);
+			});
+
+			it('should return the column', async () => {
+				const { columnBoardId } = setup();
+
+				const result = await service.findById(columnBoardId);
+
+				expect(result.id).toEqual(columnBoardId);
+			});
+		});
+	});
+
+	describe('getBoardAuthorizable', () => {
+		describe('when having an empty board', () => {
+			const setup = () => {
+				const board = columnBoardFactory.build();
+				return { board };
+			};
+
+			it('should return an empty usergroup', async () => {
+				const { board } = setup();
+				drawingDoRepo.findById.mockResolvedValueOnce(board);
+
+				const userGroup = await service.getBoardAuthorizable(board);
+
+				expect(userGroup.id).toEqual(board.id);
+				expect(userGroup.users.length).toEqual(0);
+			});
+		});
+
+		describe('when having a board with a teacher and some students', () => {
+			const setup = () => {
+				const roles = roleFactory.buildList(1, {});
+				const teacher = userFactory.buildWithId({ roles });
+				const substitutionTeacher = userFactory.buildWithId({ roles });
+				const students = userFactory.buildListWithId(3);
+				const course = courseFactory.buildWithId({
+					teachers: [teacher],
+					substitutionTeachers: [substitutionTeacher],
+					students,
+				});
+				const board = columnBoardFactory.build({ context: { type: BoardExternalReferenceType.Course, id: course.id } });
+				drawingDoRepo.findById.mockResolvedValueOnce(board);
+				courseRepo.findById.mockResolvedValueOnce(course);
+				return {
+					board,
+					teacherId: teacher.id,
+					substitutionTeacherId: substitutionTeacher.id,
+					studentIds: students.map((s) => s.id),
+					teacher,
+					substitutionTeacher,
+					students,
+				};
+			};
+
+			it('should return the teacher and the students with correct roles', async () => {
```

Comment:
pls split this into clear individual tests, e.g. student should have bord role editor. 
And I will argue, "BordRole" is simply misleading. You do not give board role, but only for drawing files should have this role